packages/common/driver-definitions/src/storage.ts

```diff
@@ -129,8 +129,16 @@ export interface IDocumentStorageService extends Partial<IDisposable> {
      * @param count - Number of the versions to be fetched.
      * @param scenarioName - scenario in which this api is called. This will be recorded by server and would help
      *  in debugging purposes to see why this call was made.
-     */
-    getVersions(versionId: string | null, count: number, scenarioName?: string): Promise<IVersion[]>;
+     * @param bypassCache - Driver may choose to cache requests and serve data from cache. That will result in
+     *  stale info returned. Callers can disable this functionality by passing bypassCache = true and ensuring
+     *  that driver will return latest information from storage.
+     */
+    getVersions(
+        versionId: string | null,
+        count: number,
+        scenarioName?: string,
+        bypassCache?: boolean,
```

Comment:
Two observations:
1. Caller can skip calling to getVersions() and go directly to getSnapshotTree(undefined). I feel like it should also support similar flow? At the same time specifying "no cache" does not have any meaning if specific version Id is provided. It feels like better direction - disallow calling getSnapshotTree() with version = undefined. What do you think?
2. I'd prefer not a boolean, but enum. With "default" and "no cache" value for starters. I can totally see us exposing more states, like "cache only" in the future - this would be potentially useful for loading flow, where we could go with loading two containers at ones - one cache only, one no cache, and going with the one that loaded faster (it would be better than overall flow for many reasons).

Reply:
**" It feels like better direction - disallow calling getSnapshotTree() with version = undefined. What do you think?"**
I think that we should not expect people to call getSnapshotTree() inorder to get the version of the snapshot. If one does not know which version to fetch, then flow should be call getVersion and then call get snapshot tree(still 1 network call due to caching and noCache mode).
If one is calling getSnapshotTree(undefined), then that means one does not care about it being the latest. If they do care for example in refreshLatestAck scenario, then getVersions should be called first. But it is not that strong of argument and I could add bypassCache in snapshotTree too.

**At the same time specifying "no cache" does not have any meaning if specific version Id is provided.**
This will not occur if getSnapshotTree does not take bypassCache argument as mode default will be executed. So if we don't change the getSnapshotTree(), then it should be fine.


Yes, I will change it to enum.